packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -539,11 +539,10 @@ export class ProfilesCache {
     }
 
     private shouldRemoveTokenFromProfile(profile: zowe.imperative.IProfileLoaded, baseProfile: zowe.imperative.IProfileLoaded): boolean {
-        return (baseProfile?.profile?.host &&
-            baseProfile?.profile?.port &&
+        return ((baseProfile?.profile?.host || baseProfile?.profile?.port) &&
             profile?.profile?.host &&
             profile?.profile?.port &&
-            (baseProfile?.profile.host !== profile?.profile.host || baseProfile?.profile.port !== profile?.profile.port) &&
+            (baseProfile?.profile.host !== profile?.profile.host || (profile?.profile.user && profile?.profile.password)) &&
             profile?.profile.tokenType?.startsWith(zowe.imperative.SessConstants.TOKEN_TYPE_APIML)) as boolean;
```

Comment:
There are a lot of factors in this boolean logic here 😅 

Would it make sense to move this line to the top so we return false before doing the other checks? If the profile doesn't have a token type, we can immediately return false.